### ISSUE TITLE
Network with 2 images input.

### ISSUE THREAD

**Issue by filipetrocadoferreira** [BVLC/caffe#4850]
Which is the simplest way to implement a network with 2 images as inputs?

**Comment by brothertook** [comment]
Just use two ImageData layers. After that you can concatenate it if you need.

**Comment by williford** [comment]
See:
http://stackoverflow.com/questions/27436987/caffe-multiple-input-images

**Comment by uzeful** [comment]
Do you have an example prototxt?

**Comment by filipetrocadoferreira** [comment]
I've used an RGB representation (and use of discard data layer)

**Comment by brothertook** [comment]
layer {
  name: "weekly-data"
  type: "ImageData"
  top: "data_img0"
  top: "vid"
  image_data_param {
source: "train.txt"
  is_color: true
  batch_size: 15
  }
  transform_param {
   scale: 0.0039215684
   
  
  }

}

layer {
  name: "weekly-data1"
  type: "ImageData"
  top: "data_img1"
  top: "vid1"
  image_data_param {
source: "test.txt"
  is_color: true
  batch_size: 15
  }
  transform_param {
   scale: 0.0039215684
   
  
  }
 
}

layer {
  name: "concat"
  bottom: "data_img0"
  bottom: "data_img1"
  top: "data_img"
  type: "Concat"
  concat_param {
    axis: 1
  }
}

**Comment by filipetrocadoferreira** [comment]
And in inference how do you copy your data to each one of input layers(c++)

**Comment by brothertook** [comment]
source: "train.txt"
train.txt contains a list if image file name one per line.
caffe get files for that list and make blob . So I don'n have to write any c++ code to load data.

**Comment by shelhamer** [comment]
From https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:

> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.

**Comment by CHENTao-U** [comment]
how about the labels in the softmax layer? there are 2 sources of labels after the concat.. how to handle this?

regards,

**Comment by shaibagon** [comment]
@CHENTao-U Please use the [caffe-users list][1] for usage, installation, or modeling questions, or other requests for help.
You may also post questions on [stackoverflow][3], make sure you tag them with `caffe` tag.  
There is also [caffe.help][4] documenting the different layers of caffe.
Do not post such requests to Issues. Doing so interferes with the development of Caffe.

Please read the [guidelines for contributing][2] before submitting this issue.

 [1]: https://groups.google.com/forum/#!forum/caffe-users
 [2]: https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md
 [3]: http://stackoverflow.com/questions/tagged/caffe
 [4]: http://caffe.help/manual/layers.html
